core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3797,6 +3797,14 @@ public static List distinct(List list) {
     return new ArrayList<>(result);
   }
 
+  /** Support the ARRAY_REPEAT function. */
+  public static @Nullable List repeat(Object element, Object count) {
+    if (count == null) {
+      return null;
+    }
+    return Collections.nCopies((Integer) count, element);
```

Comment:
Can it be Long?

I am worried that the Integer will fail for Long.

Reply:
@JiajunBernoulli it only can be integer in spark, i comments it before.
```
 override def inputTypes: Seq[AbstractDataType] = Seq(AnyDataType, IntegerType)
```